src/token/erc20/presets/erc20votes.cairo

```diff
@@ -0,0 +1,204 @@
+            sender: ContractAddress,
+            recipient: ContractAddress,
+            amount: u256
+        ) -> bool {
+            let mut unsafe_state = ERC20::unsafe_new_contract_state();
+            let caller = starknet::get_caller_address();
+            ERC20::InternalImpl::_spend_allowance(ref unsafe_state, sender, caller, amount);
+            ERC20::InternalImpl::_transfer::<
+                ERC20VotesHooksImpl
+            >(ref unsafe_state, sender, recipient, amount);
+            true
+        }
+
+        fn approve(ref self: ContractState, spender: ContractAddress, amount: u256) -> bool {
+            let mut unsafe_state = ERC20::unsafe_new_contract_state();
+            ERC20::ERC20Impl::approve(ref unsafe_state, spender, amount)
+        }
+    }
+
+    #[external(v0)]
+    fn increase_allowance(
+        ref self: ContractState, spender: ContractAddress, added_value: u256
+    ) -> bool {
+        let mut unsafe_state = ERC20::unsafe_new_contract_state();
+        ERC20::InternalImpl::_increase_allowance(ref unsafe_state, spender, added_value)
+    }
+
+    #[external(v0)]
+    fn decrease_allowance(
+        ref self: ContractState, spender: ContractAddress, subtracted_value: u256
+    ) -> bool {
+        let mut unsafe_state = ERC20::unsafe_new_contract_state();
+        ERC20::InternalImpl::_decrease_allowance(ref unsafe_state, spender, subtracted_value)
+    }
+
+    #[external(v0)]
+    impl VotesImpl of IVotes<ContractState> {
+        fn get_votes(self: @ContractState, account: ContractAddress) -> u256 {
+            let unsafe_state = ERC20Votes::unsafe_new_contract_state();
+            ERC20Votes::VotesImpl::get_votes(@unsafe_state, account)
+        }
+
+        fn get_past_votes(self: @ContractState, account: ContractAddress, timepoint: u64) -> u256 {
+            let unsafe_state = ERC20Votes::unsafe_new_contract_state();
+            ERC20Votes::VotesImpl::get_past_votes(@unsafe_state, account, timepoint)
+        }
+
+        fn get_past_total_supply(self: @ContractState, timepoint: u64) -> u256 {
+            let unsafe_state = ERC20Votes::unsafe_new_contract_state();
+            ERC20Votes::VotesImpl::get_past_total_supply(@unsafe_state, timepoint)
+        }
+
+        fn delegates(self: @ContractState, account: ContractAddress) -> ContractAddress {
+            let unsafe_state = ERC20Votes::unsafe_new_contract_state();
+            ERC20Votes::VotesImpl::delegates(@unsafe_state, account)
+        }
+
+        fn delegate(ref self: ContractState, delegatee: ContractAddress) {
+            let mut unsafe_state = ERC20Votes::unsafe_new_contract_state();
+            ERC20Votes::VotesImpl::delegate(ref unsafe_state, delegatee);
+        }
+
+        fn delegate_by_sig(
+            ref self: ContractState,
+            delegator: ContractAddress,
+            delegatee: ContractAddress,
+            nonce: felt252,
+            expiry: u64,
+            signature: Array<felt252>
```

Comment:
why `Array` and not `Span` as in account contracts? this was asked by Starkware too

Reply:
to be compliant with our Account, which didn't have have `Span` de/coding so it wasn't an option back then